torch/distributed/rpc/api.py

```diff
@@ -120,10 +206,14 @@ def shutdown(graceful=True):
         >>> rpc.shutdown()
     """
     global _agent
-    if _agent:
-        if graceful:
-            _wait_all_workers()
+
+    if graceful:
+        _wait_all_workers()
+        _agent.join()
+    try:
+        # This might raise a `TORCH_CHECK()` exception.
         _destroy_rref_context(_ignore_rref_leak)
+    finally:
```

Comment:
@mrshenli This fix the destruction segfault in Python3.5. We need to use finally clause to ensure running `_cleanup_python_rpc_handler()`

Reply:
Thanks! would you please add comments for finally here so that we can understand it in the future

Reply:
@zhaojuanmao Updated the comment.